pixl_dcmd/src/pixl_dcmd/main.py

```diff
@@ -81,344 +81,139 @@ def anonymise_dicom(dataset: Dataset) -> Dataset:
 
-def enforce_whitelist(dataset: Dataset, tags: list[dict]) -> Dataset:
-    """Delete any tags not in the tagging scheme."""
-    # For every element:
-    logger.debug("Enforcing whitelist")
-    for de in dataset:
-        keep_el = False
-        # For every entry in the YAML:
-        for i in range(len(tags)):
-            grp = tags[i]["group"]
-            el = tags[i]["element"]
-            op = tags[i]["op"]
-
-            if de.tag.group == grp and de.tag.element == el:
-                if op != "delete":
-                    keep_el = True
-
-        if not keep_el:
-            del_grp = de.tag.group
-            del_el = de.tag.element
-
-            del dataset[del_grp, del_el]
-            message = "Whitelist - deleting: {name} (0x{grp:04x},0x{el:04x})".format(
-                name=de.keyword, grp=del_grp, el=del_el
+    tag_actions = {}
+    for tag in tags_list:
+        group_el = (tag["group"], tag["element"])
+        if tag["op"] == "secure-hash":
+            tag_actions[group_el] = lambda _dataset, _tag: _secure_hash(
+                _dataset, _tag, mrn, accession_number
             )
-            logger.debug(f"\t{message}")
+            continue
+        tag_actions[group_el] = actions_map_name_functions[tag["op"]]
 
-    return dataset
+    return tag_actions
 
 
-def apply_tag_scheme(dataset: Dataset, tags: list[dict]) -> Dataset:
+def _secure_hash(dataset: Dataset, tag: tuple, mrn: str, accession_number: str) -> None:
     """
-    Apply anonymisation operations for a given set of tags to a dataset.
-    The original study time is kept before any operations are applied.
-    For example: orig_study_time = `dataset[0x0008, 0x0030].value`
+    Use the hasher API to consistently but securely hash ids later used for linking.
     """
-    logger.debug("Applying tag scheme")
+    grp = tag[0]
+    el = tag[1]
 
-    mrn = dataset[0x0010, 0x0020].value  # Patient ID
-    accession_number = dataset[0x0008, 0x0050].value  # Accession Number
+    if tag in dataset:
+        message = f"Securely hashing: (0x{grp:04x},0x{el:04x})"
+        logger.debug(f"\t{message}")
+        if grp == 0x0010 and el == 0x0020:  # Patient ID
+            pat_value = mrn + accession_number
 
-    # Set salt based on ENV VAR
-    salt_plaintext = config("SALT_VALUE")
+            hashed_value = _hash_values(pat_value)
+            # Query PIXL database
+            existing_image = query_db(mrn, accession_number)
+            # Insert the hashed_value into the PIXL database
+            add_hashed_identifier_and_save_to_db(existing_image, hashed_value)
+        elif dataset[grp, el].VR == "SH":
+            pat_value = str(dataset[grp, el].value)
+            hashed_value = _hash_values(pat_value, 16)
 
+        dataset[grp, el].value = hashed_value
+
+    else:
+        message = f"Missing linking variable (0x{grp:04x},0x{el:04x})"
+        logger.warning(f"\t{message}")
```

Comment:
this happens quite a lot, I wonder if it should also be debug

Reply:
does it? This will only be activated if there is no patient id currently